services/docker/docker-cloud-build.service.js

```diff
@@ -0,0 +1,61 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { dockerBlue, buildDockerUrl } = require('./docker-helpers')
+
+const cloudBuildSchema = Joi.object({
+  objects: Joi.array()
+    .items(Joi.object({ state: Joi.string() }).required())
+    .required(),
+}).required()
+
+module.exports = class DockerCloudBuild extends BaseJsonService {
+  async fetch({ user, repo }) {
```

Comment:
This `fetch` function can be extracted out into a single location (perhaps a DockerCloud base class) and reused instead of having it redefined in both classes

Reply:
If you don't want to create a base class, another option is to make a function:

https://github.com/badges/shields/blob/879581466cd94cbad706b92fdc5449080774fd12/services/github/github-common-fetch.js#L7-L19

One of the reasons I like using functions instead of classes to share code among services is that it makes this particularly common refactor easier to execute.